### ISSUE TITLE
esbuild EACCES issue on MacOS

### ISSUE THREAD

**Issue by mnaoumov** [mnaoumov/obsidian-fix-require-modules#5]
```
Evaluation Error: TransformError: spawnSync /Users/bpannier/Library/CloudStorage/GoogleDrive-bpannie...com/My Drive/Apps/Obsidian/DEV/Fix Require/.obsidian/plugins/fix-require-modules/node_modules/@esbuild/darwin-arm64/bin/esbuild EACCES
    at Object.spawnSync (node:internal/child_process:1124:20)
    at spawnSync (node:child_process:891:24)
    at Object.execFileSync (node:child_process:934:15)
    at Object.func [as execFileSync] (node:electron/js2c/node_init:2:2214)
    at runServiceSync (/Users/bpannier/Library/CloudStorage/GoogleDrive-bpannie...com/My Drive/Apps/Obsidian/DEV/Fix Require/.obsidian/plugins/fix-require-modules/node_modules/esbuild/lib/main.js:2084:30)
    at transformSync (/Users/bpannier/Library/CloudStorage/GoogleDrive-bpannie../My Drive/Apps/Obsidian/DEV/Fix Require/.obsidian/plugins/fix-require-modules/node_modules/esbuild/lib/main.js:1908:3)
    at eval (plugin:fix-require-modules:76522:45)
    at applyTransformersSync (plugin:fix-require-modules:76487:16)
    at transformSync (plugin:fix-require-modules:76518:22)
    at Object.transformer [as .js] (plugin:fix-require-modules:76659:18)
```

_Originally posted by @bpannier in https://github.com/mnaoumov/obsidian-fix-require-modules/issues/4#issuecomment-2453111718_

**Comment by mnaoumov** [comment]
@bpannier, I don't have a MacOS device handy. Are you willing to help me to identify the issue?

if you run `chmod +x path/to/vault/.obsidian/plugins/fix-require-modules/node_modules/@esbuild/darwin-arm64/bin/esbuild`

does it help?

**Comment by bpannier** [comment]
Indeed it fixed the problem, but raised a new one:

```
Evaluation Error: Error: Cannot find module '/Users/bpannier/Library/CloudStorage/GoogleDrive-bpannie...com/My Drive/Apps/Obsidian/DEV/Fix Require/ts/MyClass'
Require stack:
- /Users/bpannier/Library/CloudStorage/GoogleDrive-bpannie...com/My Drive/Apps/Obsidian/DEV/Fix Require/ts/myscript.ts?namespace=fix-require-modules
- /Applications/Obsidian.app/Contents/Resources/electron.asar/renderer/init.js
    at Module._resolveFilename (node:internal/modules/cjs/loader:1152:15)
    at a._resolveFilename (node:electron/js2c/renderer_init:2:2669)
    at Function.customResolveFilename [as _resolveFilename] (plugin:fix-require-modules:77085:40)
    at Module._load (node:internal/modules/cjs/loader:993:27)
    at c._load (node:electron/js2c/node_init:2:13801)
    at s._load (node:electron/js2c/renderer_init:2:31018)
    at Module.require (node:internal/modules/cjs/loader:1240:19)
    at customRequire (plugin:fix-require-modules:77039:26)
    at Module.patchedModuleRequire [as require] (plugin:fix-require-modules:77222:12)
    at require (node:internal/modules/helpers:179:18)
```

**Comment by mnaoumov** [comment]
Did you install the latest build?

**Comment by bpannier** [comment]
Ohhh dear, obviously not in that vault. After updating it works! Very cool! Thanks

**Comment by mnaoumov** [comment]
I did have `chmod +x` during first installation. But for you it didn't work. 

Can you please

1. remove the plugin completely
2. ensure `path/to/vault/.obsidian/plugins/fix-require-modules` is removed
3. Install the plugin
4. Observe the console log for errors

**Comment by bpannier** [comment]
Yes, there is an error coming up.

Also, when I set the script root directory in the plugin, this will only be used after a restart of Obsidian.

<img width="1232" alt="Xnapper-2024-11-02-21 20 30" src="https://github.com/user-attachments/assets/0ef1acda-6ee8-43ee-8907-e79b912ac044">
<img width="1962" alt="Xnapper-2024-11-02-21 21 32" src="https://github.com/user-attachments/assets/a21bc7b4-945b-46cb-8ad6-70630cc027e2">

**Comment by mnaoumov** [comment]
Fixed in https://github.com/mnaoumov/obsidian-fix-require-modules/releases/tag/5.1.1

@bpannier please confirm

**Comment by bpannier** [comment]
I can confirm after de-installing, a new install did not bring any error up and the right permission has been applied to esbuild.

Thanks!
